tools/gmhazard_scripts/db_creation/empirical_db/new_calc_emp_ds.py

```diff
@@ -0,0 +1,246 @@
+          0.9, 1.0, 1.25, 1.5, 2.0, 2.5, 3.0, 4.0, 5.0, 6.0, 7.5, 10.0]
+# fmt: on
+
+IM_TYPE_LIST = [
+    gc.im.IMType.PGA,
+    gc.im.IMType.pSA,
+    gc.im.IMType.Ds595,
+    gc.im.IMType.Ds575,
+    gc.im.IMType.AI,
+    gc.im.IMType.CAV,
+    gc.im.IMType.PGV,
+]
+
+MAG = [
+    5.25,
+    5.75,
+    6.25,
+    6.75,
+    7.25,
+    8.0,
+]  # Definitions for the maximum magnitude / rjb relation
+DIST = [125, 150, 175, 200, 250, 300]
+
+
+def get_rupture_context_df(
+    distance_store: gc.dbs.SiteSourceDB,
+    fault_df: pd.DataFrame,
+    site: str,
+    site_data: pd.Series,
+    nhm_data: pd.DataFrame,
+    tect_type: str,
+):
+    """Creating the form of dataframe for the vectorized OQ Wrapper
+    By combining site, distance and rupture information
+    """
+    distance_df = fault_df.merge(
+        distance_store.station_data(site), left_on="fault_name", right_index=True,
+    )
+    rupture_df = nhm_data.merge(
+        distance_df, left_on="fault_name", right_on="fault_name"
+    )
+    max_dist = np.minimum(
+        np.interp(rupture_df.mag, MAG, DIST),
+        common.get_max_dist_zfac_scaled(site_data),
+    )
+    rupture_df = rupture_df[rupture_df["rjb"] < max_dist]
+
+    # Adding missing columns
+    # Site Parameters
+    rupture_df["vs30"] = (
+        classdef.VS30_DEFAULT
+        if np.isnan(site_data.get("vs30", default=np.nan))
+        else float(site_data.vs30)
+    )
+    rupture_df["vs30measured"] = site_data.get("vs30measured", default=False)
+    rupture_df["z1pt0"] = (
+        classdef.estimate_z1p0(site_data.vs30)
+        if np.isnan(float(site_data.get("z1p0", default=np.nan)))
+        else float(site_data.z1p0)
+    )
+    rupture_df["z2pt5"] = (
+        classdef.estimate_z2p5(z1p0=rupture_df["z1pt0"].iloc[0])
+        if np.isnan(float(site_data.get("z2p5", default=np.nan)))
+        else float(site_data.z2p5)
+    )
+
+    # Rupture Parameter
+    rupture_df[["hypo_depth", "ztor"]] = rupture_df[["dbot", "dtop"]]
+
+    # Distance Parameter - OQ uses ry0 term
+    rupture_df[["rx", "ry0"]] = rupture_df[["rx", "ry"]].fillna(0)
+
+    return rupture_df.loc[rupture_df["tect_type"] == tect_type]
+
+
+def calculate_ds(
+    background_sources_ffp: str,
+    site_source_db_ffp: str,
+    vs30_ffp: str,
+    output_dir: str,
+    z_ffp: Optional[str],
+    ims: Optional[List[gc.im.IMType]],
+    psa_periods: Optional[List[float]],
+    model_dict_ffp: Optional[str],
+    suffix: Optional[str] = None,
+):
+    nhm_data = gc.utils.ds_nhm_to_rup_df(background_sources_ffp)
+    rupture_df = pd.DataFrame(nhm_data["rupture_name"])
+    imdb_dict, _ = common.open_imdbs(
+        model_dict_ffp, output_dir, gc.constants.SourceType.distributed, suffix=suffix,
+    )
+    model_dict = empirical_factory.read_model_dict(model_dict_ffp)
+
+    with gc.dbs.SiteSourceDB(site_source_db_ffp) as distance_store:
+        fault_df, _, site_df, _ = common.get_work(
```

Comment:
Since neither ds or flt use MPI I think we should rename the `get_work` function and remove the MPI components, i.e. `rank`, `size` (inputs) and `work` (outputs).

Reply:
I think this can be done but it's kinda related to the reason I set `new_calc_emp_ds.py` instead of modifying the `calc_emp_ds`.

we probably need to confirm but as long as we need all the models that the current config has. E.g., K_20, ZA_06, CB_14 and ASK_14. (K_20 and ZA_06 does not support the standard periods, CB_14 and ASK_14, we need to implement width estimation), we should still need to use the `calc_emp_ds`

As the existing `calc_emp_ds` works, we just cannot run with many processes so until we move on to the new_calc_emp_ds, we might have to keep the `get_work` as is for now? otherwise can be done, but I guess it might be better to do it in another PR to simplify `get_work` then apply those changes to two files(ds and flt)

Reply:
After this is merged, until we sort out periods and width estimation there still is a use case for the old workflow. But once we sort those issues, we can edit the get_work function and delete the old workflow.